TestON/drivers/common/cli/emulator/mininetclidriver.py

```diff
@@ -1377,20 +1377,30 @@ def stopNet( self, fileName = "", timeout=5):
             response = main.FALSE
         return response
 
-    def arping( self, src, dest, destmac ):
-        self.handle.sendline( '' )
-        self.handle.expect( [ "mininet", pexpect.EOF, pexpect.TIMEOUT ] )
-
-        self.handle.sendline( src + ' arping ' + dest )
+    def arping( self, host="", ip="10.128.20.211" ):
+        """
+        Description:
+            Sends arp message from mininet host for hosts discovery
+        Required:
+            host - hosts name
+        Optional:
+            ip - ip address that does not exist in the network so there would
+                 be no reply.
```

Comment:
This one im still confuse, even though the arping does not recieve reply"
it will still discover hosts. Should I re.search for a specific output?

On Tue, May 19, 2015 at 11:41 AM, Jon Hall <notifications@github.com> wrote:

> In TestON/drivers/common/cli/emulator/mininetclidriver.py
> <https://github.com/opennetworkinglab/ONLabTest/pull/117#discussion_r30631966>
> :
>
> > @@ -1377,20 +1377,30 @@ def stopNet( self, fileName = "", timeout=5):
> >              response = main.FALSE
> >          return response
> >
> > -    def arping( self, src, dest, destmac ):
> > -        self.handle.sendline( '' )
> > -        self.handle.expect( [ "mininet", pexpect.EOF, pexpect.TIMEOUT ] )
> > -
> > -        self.handle.sendline( src + ' arping ' + dest )
> > +    def arping( self, host="", ip="10.128.20.211" ):
> > +        """
> > +        Description:
> > +            Sends arp message from mininet host for hosts discovery
> > +        Required:
> > +            host - hosts name
> > +        Optional:
> > +            ip - ip address that does not exist in the network so there would
> > +                 be no reply.
>
> Returns:
> main.TRUE if arping is sent
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/opennetworkinglab/ONLabTest/pull/117/files#r30631966>.
>